cylc/uiserver/scripts/gui.py

```diff
@@ -20,10 +20,163 @@
 For a multi-user system see `cylc hub`.
 """
 
+from ansimarkup import parse as cparse
+import argparse
+import asyncio
+from contextlib import suppress
+from glob import glob
+import os
+from pathlib import Path
+import random
+import re
+import sys
+import webbrowser
+
+
+from cylc.flow.id_cli import parse_id_async
+from cylc.flow.exceptions import (
+    InputError,
+    WorkflowFilesError
+)
+
 from cylc.uiserver import init_log
-from cylc.uiserver.app import CylcUIServer
+from cylc.uiserver.app import (
+    CylcUIServer,
+    INFO_FILES_DIR
+)
+
+CLI_OPT_NEW = "--new"
 
 
 def main(*argv):
     init_log()
-    return CylcUIServer.launch_instance(argv or None)
+    new_gui = False
+    new_gui = CLI_OPT_NEW in argv
+    jp_server_opts, new_gui, workflow_id = parse_args_opts()
+    if '--help' not in sys.argv:
+        # get existing jpserver-<pid>-open.html files
+        # assume if this exists then the server is still running
+        # these files are cleaned by jpserver on shutdown
+        existing_guis = glob(os.path.join(INFO_FILES_DIR, "*open.html"))
+        if existing_guis and not new_gui:
+            gui_file = random.choice(existing_guis)
+            print(
+                "Opening with existing gui." +
+                f" Use {CLI_OPT_NEW} option for a new gui.",
+                file=sys.stderr
+            )
+            update_html_file(gui_file, workflow_id)
+            if '--no-browser' not in sys.argv:
+                webbrowser.open(gui_file, autoraise=True)
+            return
+    return CylcUIServer.launch_instance(
+        jp_server_opts or None, workflow_id=workflow_id
+    )
+
+
+def raise_error(error: str, msg: str):
```

Comment:
I would call this `print_error` seeing as it doesn't actually raise an exception